### DISCUSSION TITLE
Looking for a way to set network interface MAC address

### DISCUSSION THREAD

**Discussion by littlejackal** [siderolabs/talos#5639] [Q&A]
I've been hoping to deploy a tiny Talos cluster with a handful of Rock64 boards, since Talos has an official build for this cheap SBC.

Unfortunately, and I'm not certain whether this is all Rock64 boards or just my batch, this particular series of SBC seems to bring up the interface with an identical MAC address. In my case, this happens to be ea:77:79:99:d4:1e. 

On other Linux distributions, it seems I can set the MAC either through boot config files, or on the system itself using any number of conventional means, but none of these are available to me on Talos.

So right now I am stuck only being able to use a single Rock64. I can configure Talos on one, but as soon as another comes online there is a duplicate ea:77:79:99:d4:1e on the network and the first drops off.

This is more of a Rock64 problem than a Talos problem, but it would be nice to have a way for the OS to force a MAC for a specific interface.

Ideally I'd be able to apply a machine config along the lines of:

```yaml
interfaces:
      - interface: eth0
        addresses:
          - 10.0.80.16/24
        mtu: 1500
        hwaddr: ea:77:79:99:d4:2e
```

and Talos would dutifully do what it needs to do behind the scenes.

Short of modifying Talos itself, as an UGLY hack, I'm curious whether there's any way to have it run the equivalent of `ifconfig eth0 hw ether ea:77:79:99:d4:2e` behind the scenes, using any binaries it might have available to itself?

**Comment by frezbo** [comment]
you could possibly do that from a kubernetes pod running in hostnetwork mode, but it will not survive a reboot. By any chance are you using the u-boot from rockpi? The u-boot provided by talos for the rockpi's probably have the fix for the stable mac address
